src/utility/diffable_threads.jl

```diff
@@ -0,0 +1,71 @@
+"""
+    dtmap(args...; kwargs...)
+
+Differentiable wrapper around `OhMyThreads.tmap`.
+
+All calls of `dtmap` inside of PEPSKit use the threading keyword arguments stored
+inside `Default.threading_kwargs` which can be modified using `set_threading_kwargs!`.
+"""
+dtmap(args...; kwargs...) = tmap(args...; kwargs...)
+
+# Follows the `map` rrule from ChainRules.jl but specified for the case of one AbstractArray that is being mapped
+# https://github.com/JuliaDiff/ChainRules.jl/blob/e245d50a1ae56ce46fc8c1f0fe9b925964f1146e/src/rulesets/Base/base.jl#L243
+function ChainRulesCore.rrule(
+    config::RuleConfig{>:HasReverseMode}, ::typeof(dtmap), f, A::AbstractArray; kwargs...
+)
+    el_rrules = tmap(A; kwargs...) do a
+        rrule_via_ad(config, f, a)
+    end
+    y = map(first, el_rrules)
+    function dtmap_pullback(dy_raw)
+        dy = unthunk(dy_raw)
+        backevals = tmap(CartesianIndices(A); kwargs...) do idx
+            last(el_rrules[idx])(dy[idx])
+        end
+        df = ProjectTo(f)(sum(first, backevals))
+        dA = tmap(CartesianIndices(A); kwargs...) do idx
```

Comment:
Here, I don't think it is really worth it to use `tmap`, there are no more computations (I think), it should just be unpacking the values...

Reply:
Oh yes definitely, probably creates more overhead

Reply:
> I think I should have elaborated a bit more on what I had in mind with the scoped values: I would keep the threading strategies in scoped values, such that they can always be accessed as if they are global values. (this prevents them from bloating all of our algorithms) The benefit of having them as a scoped value however means that users could still change them by calling the peps function from within a scope with a modified scoped value, thus changing the scheduler.

Alright, thanks for clarifying! The way the threading kwargs are defined should be fine anyway, right? I like having them inside the `Defaults`